a3p-integration/proposals/z:acceptance/auction.test.js

```diff
@@ -0,0 +1,389 @@
+
+const ambientAuthority = {
+  query: agd.query,
+  follow: agoric.follow,
+  setTimeout: globalThis.setTimeout,
+};
+
+const fromBoard = makeFromBoard();
+const marshaller = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+
+export const bankSend = (from, addr, wanted) => {
+  const chain = ['--chain-id', CHAINID];
+  const fromArg = ['--from', from];
+  const testKeyring = ['--keyring-backend', 'test'];
+  const noise = [...fromArg, ...chain, ...testKeyring, '--yes'];
+
+  return agd.tx('bank', 'send', from, addr, wanted, ...noise);
+};
+
+const config = {
+  depositor: {
+    depositValue: '100000000',
+    offerId: `gov1-deposit-${Date.now()}`,
+  },
+  price: 50.0,
+  bidsSetup: [
+    {
+      bidder: USER1ADDR,
+      bidderFund: {
+        value: 90000000,
+        denom: 'uist',
+      },
+      offerId: `user1-bid-${Date.now()}`,
+      give: '90IST',
+      price: 46,
+    },
+    {
+      bidder: GOV3ADDR,
+      bidderFund: {
+        value: 150000000,
+        denom: 'uist',
+      },
+      offerId: `gov3-bid-${Date.now()}`,
+      give: '150IST',
+      discount: '13',
+    },
+  ],
+  bidsOutcome: [
+    {
+      payouts: {
+        Bid: 0,
+        Collateral: 1.68421,
+      },
+    },
+    {
+      payouts: {
+        Bid: 0,
+        Collateral: 2.0,
+      },
+    },
+    {
+      payouts: {
+        Bid: 0,
+        Collateral: 3.448275,
+      },
+    },
+  ],
+};
+
+const pushPricesForAuction = async t => {
+  const oraclesByBrand = new Map();
+  await addPreexistingOracles('ATOM', oraclesByBrand);
+
+  await pushPrices(config.price, 'ATOM', oraclesByBrand, t.context.roundId + 1);
+
+  await retryUntilCondition(
+    () => getPriceQuote('ATOM'),
+    res => res === `+${scale6(config.price).toString()}`,
+    'price not pushed yet',
+    {
+      log: t.log,
+      setTimeout: globalThis.setTimeout,
+      ...t.context.pushPriceRetryOpts,
+    },
+  );
+};
+
+const fundAccts = async (depositorAmt = '100000000', t) => {
+  const retryOpts = t.context.retryOpts.bankSendRetryOpts;
+
+  await bankSend(VALIDATORADDR, GOV1ADDR, `${depositorAmt}${ATOM_DENOM}`),
+    await waitUntilAccountFunded(
+      GOV1ADDR,
+      ambientAuthority,
+      { denom: ATOM_DENOM, value: Number(depositorAmt) },
+      { errorMessage: 'gov1 not funded yet', ...retryOpts },
+    );
+
+  const user1Fund = config.bidsSetup[0].bidderFund;
+  await bankSend(GOV1ADDR, USER1ADDR, `${user1Fund.value}${user1Fund.denom}`);
+  await waitUntilAccountFunded(USER1ADDR, ambientAuthority, user1Fund, {
+    errorMessage: 'user1 not funded yet',
+    ...retryOpts,
+  });
+
+  const gov3Fund = config.bidsSetup[1].bidderFund;
+  await bankSend(GOV1ADDR, GOV3ADDR, `${gov3Fund.value}${gov3Fund.denom}`);
+  await waitUntilAccountFunded(GOV3ADDR, ambientAuthority, gov3Fund, {
+    errorMessage: 'gov3 not funded yet',
+    ...retryOpts,
+  });
+};
+
+const bidByPrice = (price, give, offerId, bidder, t) => {
```

Comment:
please move the helper methods to a separate file like auction-lib.js in the same directory.

Reply:
Thinking about [test-lib](https://github.com/Agoric/agoric-sdk/tree/anil/auction-tests/a3p-integration/proposals/z%3Aacceptance/test-lib) directory for this. While at it, what do you think if we move [vaults.mts](https://github.com/Agoric/agoric-sdk/blob/anil/auction-tests/a3p-integration/proposals/z%3Aacceptance/lib/vaults.mts) to `test-lib` and get rid of `lib` directory? Having two directories that have "*lib" in their name seems kind of confusing to me.

Reply:
Good idea, since it's used only in testing.

In some other images a vaults module may be needed for the `use` image.

https://github.com/Agoric/agoric-3-proposals/issues/115 isn't implemented yet but eventually we'l want some consistent filenames to avoid invalidating build stages unnecessarily.

Reply:
I don't mind having a consistent directory for the lib files. An issue for me was that they have to be duplicated in each proposal, since the proposal directories are mounted separately when running proposals.

Reply:
I have moved `vaults.mts` to `test-lib` and removed `lib` in https://github.com/Agoric/agoric-sdk/pull/10229/commits/50a0e17318c083daea6fc2628216adf294525e7a. Resolving this one for now.